webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/Score.scala

```diff
@@ -0,0 +1,251 @@
+  private[this] val scoreCache: Ref[Map[NodeId, List[Score[_]]]] = scoreRepository.getAll().flatMap(Ref.make(_)).runNow
+
+  def getAll():                       IOResult[Map[NodeId, GlobalScore]] = cache.get
+  def getGlobalScore(nodeId: NodeId): IOResult[GlobalScore]              = {
+    for {
+      c   <- cache.get
+      res <-
+        c.get(nodeId) match {
+          case Some(g) => g.succeed
+          case None    => Inconsistency("No global score for node").fail
+        }
+    } yield {
+      res
+    }
+  }
+
+  def getScoreDetails(nodeId: NodeId): IOResult[List[Score[_]]] = {
+    for {
+      c   <- scoreCache.get
+      res <-
+        c.get(nodeId) match {
+          case Some(g) => g.succeed
+          case None    => Inconsistency("No score for node").fail
+        }
+    } yield {
+      res
+    }
+  }
+
+  def cleanScore(name: String) = {
+    for {
+      _ <- cache.update(_.map { case (id, gscore) => (id, gscore.copy(details = gscore.details.filterNot(_.name == name))) })
+    } yield {}
+  }
+
+  def update(newScores: Map[NodeId, List[Score[_]]]) = {
+    for {
+      c           <- cache.get
+      // sc <- scoreCache.get
+      updatedValue = (for {
+                       (nodeId, newScores) <- newScores
+                     } yield {
+                       val oldScores = c.get(nodeId) match {
+                         case None           => Nil
+                         case Some(oldScore) => oldScore.details
+                       }
+                       (nodeId, GlobalScoreService.computeGlobalScore(oldScores, newScores))
+                     })
+
+      updateScoreCache <- ZIO.foreach(newScores.toList) {
+                            case (nodeId, scores) =>
+                              ZIO.foreach(scores)(score => {
+                                scoreRepository.saveScore(nodeId, score).catchAll(err => ReportLoggerPure.info(err.fullMsg)) *>
+                                scoreCache.update(sc =>
+                                  sc + ((nodeId, score :: sc.get(nodeId).getOrElse(Nil).filter(_.name != score.name)))
+                                )
+                              })
+
+                          }
+      updatedCache     <- ZIO.foreach(updatedValue.toList) {
+                            case (nodeId, score) => globalScoreRepository.save(nodeId, score) *> cache.update(_.+((nodeId, score)))
+                          }
+    } yield {}
+
+  }
+}
+
+class ScoreServiceManager(readScore: ScoreService) {
+
+  val handlers: Ref[List[ScoreEventHandler]] =
+    Ref.make(ComplianceScoreEventHandler :: InventoryEventHandler :: List.empty[ScoreEventHandler]).runNow
+
+  def registerHandler(handler: ScoreEventHandler) = {
+    handlers.update(handler :: _)
+  }
+
+  def handleEvent(scoreEvent: ScoreEvent) = {
+    (for {
+      h       <- handlers.get
+      _       <- ReportLoggerPure.info(s"new event ${scoreEvent}")
```

Comment:
the need to be trace level (or debug perhaps?).
Perhaps it need its own `ScoreLoggerPure` too, we will likely need it to investigate strange computation results or things like that

Reply:
debug

Reply:
There is a lot of debug output that still needs to be removed, and tests not passing. told you this pr was not ready yet and UI PR needs to be merged before ;)